docs/specs/balance-fetcher.md

```diff
@@ -0,0 +1,311 @@
+# Balance Fetcher Technical Specification
+
+## 1. Overview
+
+The Balance Fetcher is a critical component in the AEP Fee Calculator system that retrieves and stores historical balance information for reward distributors on the Arbitrum Nova network. It queries blockchain state at specific end-of-day blocks to capture point-in-time balance snapshots necessary for accurate fee calculations.
+
+## 2. Purpose
+
+### Primary Objectives
+
+- Retrieve end-of-day ETH balances for all tracked reward distributors
+- Store historical balance data in a consistent, accessible format
+- Support incremental processing to handle new dates efficiently
+- Handle distributors created mid-period by only fetching balances from their creation date onward
+
+### Component Responsibilities
+
+- Query blockchain state using RPC calls to retrieve balances
+- Manage balance data persistence for each distributor
+- Ensure data integrity and handle RPC failures gracefully
+- Maintain idempotent operations for reliability
+
+### Success Criteria
+
+- All tracked distributors have balance data for all applicable dates
```

Comment:
Update everywhere in this doc to specify that we're only doing this for one distributor